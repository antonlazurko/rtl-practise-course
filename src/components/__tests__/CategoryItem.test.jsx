import { screen } from '@testing-library/react'
import { renderWithRouter } from '../../utils/testing'
import { CategoryItem } from '../CategoryItem'

describe('CategoryItem', () => {
    it('renders CategoryItem correctly', () => {
        renderWithRouter(<CategoryItem
            strCategory="Beef"
            strCategoryThumb="https://www.themealdb.com/images/category/beef.png"
            strCategoryDescription="Test Description"/>)
        expect(screen.getByRole('article')).toMatchSnapshot()
    })
})