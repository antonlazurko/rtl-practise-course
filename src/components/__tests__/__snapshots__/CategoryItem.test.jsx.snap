// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem renders CategoryItem correctly 1`] = `
<div
  class="card"
  role="article"
>
  <div
    class="card-image"
  >
    <img
      alt="Beef"
      src="https://www.themealdb.com/images/category/beef.png"
    />
  </div>
  <div
    class="card-content"
  >
    <span
      class="card-title"
    >
      Beef
    </span>
    <p>
      Test Description
      ...
    </p>
  </div>
  <div
    class="card-action"
  >
    <a
      class="btn"
      href="/category/Beef"
    >
      Watch category
    </a>
  </div>
</div>
`;
